src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/ReportFilters/ReportFilters.utils.ts

```diff
@@ -0,0 +1,26 @@
+import { AutocompleteOption } from 'shared/components/FormComponents';
+import { Identifier } from 'modules/Dashboard/features/RespondentData/RespondentData.types';
+
+export const getUniqueIdentifierOptions = (identifiers: Identifier[]) => {
+  const uniqueIdentifiersSet = new Set<string>();
+
+  return identifiers.reduce((uniqueIdentifiers: AutocompleteOption[], identifierItem) => {
+    if (!identifierItem) return uniqueIdentifiers;
+
+    const { decryptedValue } = identifierItem;
+
+    if (!uniqueIdentifiersSet.has(decryptedValue)) {
+      uniqueIdentifiersSet.add(decryptedValue);
+
+      return [
+        ...uniqueIdentifiers,
+        {
+          label: decryptedValue,
+          id: decryptedValue,
+        },
+      ];
+    }
+
+    return uniqueIdentifiers;
+  }, []);
+};
```

Comment:
```
export const getUniqueIdentifierOptions = (identifiers: Identifier[]) => {
  const uniqueIdentifiersSet = new Set<string>();
  const uniqueIdentifiers: AutocompleteOption[] = [];

  identifiers.forEach(identifierItem => {
    if (!identifierItem) return;

    const { decryptedValue } = identifierItem;

    if (!uniqueIdentifiersSet.has(decryptedValue)) {
      uniqueIdentifiersSet.add(decryptedValue);
      uniqueIdentifiers.push({
        label: decryptedValue,
        id: decryptedValue,
      });
    }
  });

  return uniqueIdentifiers;
};
```

can we use forEach instead of reduce in this case? 
since the spread operator inside the reduce creates a new array each time a unique value is added, it seems that foreach would be a little effective here?